packages/babel-parser/src/plugins/flow/index.js

```diff
@@ -1184,7 +1182,12 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           if (kind === "get" || kind === "set") {
             this.flowCheckGetterSetterParams(node);
           }
-          if (node.key.name === "constructor" && node.value.this) {
+          /** Declared classes/interfaces do not allow spread */
+          if (
+            !allowSpread &&
+            node.key.name === "constructor" &&
+            node.value.this
+          ) {
```

Comment:
🤔 `interface` does allow `this` in `constructor`: 
```
declare interface A {
    constructor(this: number): {}
}

interface B {
    constructor(this: number): {}
}
```
Is that intended? https://flow.org/try/#0CYUwxgNghgTiAEBLAdgFxDAZlMCCC8A3gFDxnxgD2yAzqjAK5iqUwAUqAFojQFzzIGAWwBGGAJT9CAX2KziKdFhwIAQkVLkqtekxbsuPfoNESps6UA

Reply:
aha no, this should not be allowed, good catch

Reply:
Just to clarify, this is a bug in the Flow implementation and not in this PR?

Reply:
yep